### PR TITLE
Closes #2825, Closes #2827 Add module installer for views_remote_data in update hook.

### LINKED ISSUE #2825: Sites with "Quickstart Event - Trellis Event Importer" prior to 2.8.0-beta1 need a database update to enable new dependency for views_data_export.
After enabling Quickstart Event - Trellis Event Importer via the UI and going to `/admin/trellis-event-importer` I see a white screen with the following error message.
```
Fatal error: Trait "Drupal\views_remote_data\Plugin\views\PropertyPluginTrait" not found in /code/web/profiles/custom/az_quickstart/modules/custom/az_event/az_event_trellis/src/Plugin/views/filter/AZEventTrellisViewsAttributeFilter.php on line 19
```

image

When I look at the info file for `az_event_trellis` i see the dependency for `views_remote_data:views_remote_data` but somehow Drupal allowed the module to be installed without enabling views_remote_data.

image

### LINKED ISSUE #2827: Standard text format settings page produces a console error related to CKEditor 5
## Problem/Motivation
During our testing on https://28-beta1-azs-quickstart.pantheonsite.io the Embed Content and Embed Blocks were both missing from the CK Editor in the Standard text formatter. 

The [Standard text format settings page](https://28-beta1-azs-quickstart.pantheonsite.io/admin/config/content/formats/manage/az_standard?destination=/admin/config/content/formats) produces a console error related to CKEditor 5:
```
drupal.js?v=10.1.5:64 Uncaught TypeError: Cannot read properties of undefined (reading ‘id’)
    at ckeditor5.admin.js?s2c3nt:568:15
    at Array.map (<anonymous>)
    at ckeditor5.admin.js?s2c3nt:565:50
    at Array.forEach (<anonymous>)
    at Object.attach (ckeditor5.admin.js?s2c3nt:537:9)
    at drupal.js?v=10.1.5:166:24
    at Array.forEach (<anonymous>)
    at Drupal.attachBehaviors (drupal.js?v=10.1.5:162:34)
    at drupal.init.js?v=10.1.5:32:12
    at HTMLDocument.listener (drupal.init.js?v=10.1.5:20:7)
```

### To Reproduce
We aren't sure what caused this problem on this site, since it seems to be working correctly on most of the other sites that we are testing on.

### Expected behavior
Installation of the 2.8 release should not remove the embedding options from the CK Editor on _any_ sites.

##  Screenshots
[Expected Toolbar](https://quickstart.arizona.edu/sites/default/files/styles/az_large/public/2022-04/wysiwyg.png?itok=N84d6Db3) Follow link and see section 9.
wysiwyg

Actual Toolbar
Editor Toolbar Labled

### PR DESCRIPTION
**Pull request by trackleft**
## Description
Add a hook to install the `views_remote_data` dependency for `az_trellis_events`.

## Related issues
Closes #2825 
Closes #2827

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [x] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [x] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/custom/az_event/az_event_trellis/az_event_trellis.install [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+/**
+ * @file
+ * Install, update and uninstall functions for az_event_trellis.
+ */
+
+/**
+ * Ensure views_data_export module is installed.
+ */
+function az_event_trellis_update_1020801() {
+  \Drupal::service('module_installer')->install(['views_data_export']);
```
Comment (trackleft) [inline modules/custom/az_event/az_event_trellis/az_event_trellis.install:12]:
```suggestion
  \Drupal::service('module_installer')->install(['views_remote_data']);
```

File: modules/custom/az_event/az_event_trellis/az_event_trellis.install [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+/**
+ * @file
+ * Install, update and uninstall functions for az_event_trellis.
+ */
+
+/**
+ * Ensure views_data_export module is installed.
```
Comment (tadean) [inline modules/custom/az_event/az_event_trellis/az_event_trellis.install:9]:
```suggestion
 * Ensure views_remote_data module is installed.
```

Comment (trackleft) [reply modules/custom/az_event/az_event_trellis/az_event_trellis.install:9]: Ha, oops

**Comment by tadean** [comment]
Tested with site originating from `tag/2.7.5`
```
$ lando drush pm-list --filter="name~=#(views_remote_data)#i"
 --------- --------------------------------------- ---------- --------- 
  Package   Name                                    Status     Version  
 --------- --------------------------------------- ---------- --------- 
  Other     Views Remote Data (views_remote_data)   Disabled   1.0.2    
 --------- --------------------------------------- ---------- ---------
 ```
 Run updates:
 ```
 $ lando drush updb -y
 ------------------ ------------------------------- --------------- -------------------------------------------------------------------------------------- 
  Module             Update ID                       Type            Description                                                                           
 ------------------ ------------------------------- --------------- -------------------------------------------------------------------------------------- 
  ...                                 
  az_event_trellis   1020801                         hook_update_n   1020801 - Ensure views_remote_data module is installed.
  ...
 ------------------ ------------------------------- --------------- -------------------------------------------------------------------------------------- 

 // Do you wish to run the specified pending updates?: yes.                                                             

...
>  [notice] Update started: az_event_trellis_update_1020801
>  [notice] Update completed: az_event_trellis_update_1020801
...
 [success] Finished performing updates.
 ```
 Check post-update status:
 ```
 $ lando drush pm-list --filter="name~=#(views_remote_data)#i"

 --------- --------------------------------------- --------- --------- 
  Package   Name                                    Status    Version  
 --------- --------------------------------------- --------- --------- 
  Other     Views Remote Data (views_remote_data)   Enabled   1.0.2    
 --------- --------------------------------------- --------- ---------
 ```

**Review by tadean — APPROVED** [review]

**Review by joeparsons — APPROVED** [review]

**Review by camikazegreen — APPROVED** [review]

**Pull request merged**